Entities/GlassBerry.cs

```diff
@@ -0,0 +1,376 @@
+                    return;
+                wasBroken = true;
+                p.Leader.LoseFollower(Follower);
+                Add(new Coroutine(GlassBreakReturnRoutine(), true));
+            }
+        }
+
+        // Routine to "break" the berry and then respawn it
+        private IEnumerator GlassBreakReturnRoutine() {
+            Collidable = false;
+            sprite.Scale = Vector2.One * 2f;
+            Audio.Play("event:/game/general/diamond_touch", Position);
+            yield return 0.05f;
+
+
+            Input.Rumble(RumbleStrength.Medium, RumbleLength.Medium);
+
+            // Add obvious "shard" lines
+            for (int i = 0; i < 12; i++) {
+                float dir = Calc.Random.NextFloat(6.2831855f);
+                base.SceneAs<Level>().ParticlesFG.Emit(StrawberrySeed.P_Burst, 1, Position + Calc.AngleToVector(dir, 4f), Vector2.Zero, dir);
+            }
+
+            SceneAs<Level>().Displacement.AddBurst(Position, 0.2f, 8f, 28f, 0.2f, null, null);
+            Visible = false;
+            Position = start;
+            yield return 1.5f;
+
+
+            Audio.Play("event:/game/general/seed_reappear", Position);
+            sprite.Scale = Vector2.One;
+            Visible = true;
+            Collidable = true;
+            wasBroken = false;
+            SceneAs<Level>().Displacement.AddBurst(Position, 0.2f, 8f, 28f, 0.2f, null, null);
+            yield break;
+        }
+
+        // Requested implementation for using IStrawberry.
+        public void OnCollect() {
+            // Bail if we're already "collected".
+            if (collected)
+                return;
+
+            collected = true;
+
+            // This is used for the ascending "score" effect and even leads to 1UPs!
+            int collectIndex = 0;
+
+            if (Follower.Leader != null) {
+                Player player = Follower.Leader.Entity as Player;
+                collectIndex = player.StrawberryCollectIndex;
+                player.StrawberryCollectIndex++;
+                player.StrawberryCollectResetTimer = 2.5f;
+                Follower.Leader.LoseFollower(Follower);
+            }
+
+            // Save the Strawberry. It's not a "secret" berry, so let's not say it's "golden" for the savedata.
+            SaveData.Instance.AddStrawberry(ID, false);
+
+            // Make the Strawberry not load any more in this Session.
+            Session session = SceneAs<Level>().Session;
+            session.DoNotLoad.Add(ID);
+            session.Strawberries.Add(ID);
+            session.UpdateLevelStartDashes();
+
+            // Coroutines allow certain processes to run independently, so to speak.
+            Add(new Coroutine(CollectRoutine(collectIndex), true));
+        }
+
+        private IEnumerator CollectRoutine(int collectIndex) {
+            Level level = base.Scene as Level;
+            base.Tag = Tags.TransitionUpdate;
+            base.Depth = -2000010;
```

Comment:
The `base` accesses on this and the preceding lines don't seem necessary. Also, `Scene as Level` isn't consistent with the previous `SceneAs<Level>`.